### ISSUE TITLE
KA-Lite Exercises & Videos

### ISSUE THREAD

**Issue by j-schwartz** [rachelproject/rachelplus#3]
KA-Lite exercises.  I have downloaded them separately as native download fails.  Unsure which files I need to put where

**Comment by spatiald** [comment]
Do you have them available for download from somewhere?  Did you pull them from kalite?

**Comment by j-schwartz** [comment]
I can make them available. They are available from kalite directly though.
You have to find the http address which appears briefly during ka lite
install. Dropping it in a browser will enable a download. Jamie at ka lite
said the issue will be fixed by end of next week.

**Comment by spatiald** [comment]
So this was an issue with the Pi versions...with KA Lite installed, we were duplicating video content.  We had scripts to fix that and symbolically link the USB video files to KA Lite's.  This way, we only maintained one set on disk.

So, here a couple KA Lite questions I need help with:
- @j-schwartz - Is there a index.php module built with the shell redirect to :8008 for KA Lite?
- Are we already sym linking the files and I don't know about it?
- Do we want to sym link them?  (I think the answer is yes)
- I am still unsure about the best way to import the lessons...we have the following files (some I know where to put but not what they do; see the ka-lite install portion of the cap-rachel-configure.sh script) - assessmentitems.json (GitHub), ka-lite_content.zip (rachelfriends.org)...then there is also data-khan.zip (rachelfriends.org) and assessment.zip (rachelfriends.org) which I am unsure of.

Thanks for your help!
sam

**Comment by bnordha** [comment]
@spatiald

 

Regarding the files, things are confusing, but here is what I have found so far.

 

1)      ka-lite_content.zip contains the assessment and most resized video files.

2)      data-khan.zip contains images and assessmentitems.json but only assessmentitems.json is needed 

3)      assessment.zip is from ka-lite and contains the assessments but is missing the assessmentitems.json  file which is needed to unzip the file according to the ka-lite documentation. (https://github.com/learningequality/ka-lite/wiki/Getting-started)

 

Thus, at this point only assessmentitems.json  and data-khan.zip are required. However, there are some video files missing and I have not fully tested whether all assessments and exercises are there.

 

Bernd

**Comment by edresor** [comment]
GETTING ALL OF THE EXERCISES TO WORK ON THE INTEL CAP:

Bernd and others,

Thanks for your example and your help.

I finally got all of the exercises to work as far as I can tell.

I tried the instructions from Learning Equality with the assessments.zip archive downloaded both directly from Learning Equality and also from RachelFriends.org without any success.  (These folders may be helpful from  with KA-Lite Version 0.14.  A lot of work is going into creating exercises using a the Perseus platform developed by Khan Academy.)

I believe my copy of assessmentitems.json is from khan folder in ka-lite_content.zip or from data-khan.zip both at RachelFriends.org.  If you do not have this folder already, data-khan.zip is the quickest way to get it, but data-khan.zip does not include the assessmentitems.json.version file.  You can create it with Notepad or another text editor.  The file only needs one line with 0.13 in that line.

I have compared the assessmentitems.json files from both of these sources and they appear to be identical.  (I have not figured out what to do with the more 20,000 other folders in the khan folder in ka-lite_content.zip.)

The instructions below might help especially for beginners who make the simple mistakes which I have made.   [A few of my mistakes are noted in brackets.]

Preparation:
Make an archive named:  assessment.zip  [Do not use the name: assessmentitems.zip]
Include these two files (See information below on the files which I am using) :
    1.  assessmentitems.json
    2.  assessmentitems.json.version

To use my commands as written below, copy assessment.zip into:
       /media/RACHEL/kacontent/ on the CAP.

Commands:

cap@WRTD-303N-Server:~$ cd ka-lite

[I could not run kalite from the root directory.  I use kalite status to see if I have kalite working.  Remember to use sudo.]

cap@WRTD-303N-Server:~/ka-lite$ sudo bin/kalite manage unpack_assessment_zip /media/RACHEL/kacontent/assessme
ent.zip -f
cap@WRTD-303N-Server:~/ka-lite$ sudo bin/kalite restart

[If you don't restart, you may think you have failed even when you have finally succeeded.]

The two files are actually copied into:
    /root/ka-lite/content/khan/

The kalite management program for KA-Lite does not provide clear error messages, so you need to remember to use sudo, etc.  Also you can use the status command:  kalite status to check that everything is working properly.

A good way to check if the exercises are working is to check to see if all of the exercises work in:
      Math > Early Marth > Counting > Counting.
     Science > Chemistry > Chemical reactions and stoichiometry > Balancing chemical equations

***************************
INFORMATION ON THE FILES THAT I AM USING (from Windows Explore > Properties)

assessmentitems.json   48,024,106 bytes  Modified: ?Saturday, ?May ?30, ?2015, ??3:25:29 PM  Created and Accessed: ?Saturday, ?June ?20, ?2015, ??2:56:33 PM
assessmentitems.json.version  4 bytes    similar dates to the above file.  Contents: 0.13

  The file begins and ends as follows:
BEGINS:
{"xfc05e430f582a38d": {"kind": "AssessmentItem", "name": "New type 1 10", "title": "New type 1 10", "slug": "xfc05e430f582a38d", "author_names": ["Sharon F."], "item_data": "{\"question\":{\"content\":

\"**Move the orange dot to.......

ENDS:
..........The translocation of chromosomes 14 and 18 placed the bcl-2 gene near a strong promoter. The translocation places Bcl-2 right after the promoter for the Ig heavy chain gene. This heavy chain is up 

regulated in B-cells, so as a consequence in translocated chromosomes, bcl-2 is translated more often than normal.\",\"images\":{},\"widgets\":{}}]}", "sha": "4d6fce652f32d9eefcdcb999f3d41bcc511b5cd0", 

"path": "xb8d656f0f0e31ada/", "id": "xb8d656f0f0e31ada"}}

**Comment by bnordha** [comment]
@j-schwartz

 

Jeremy

 

We all have lots of issues with the videos and exercises of ka-lite. Would it be better if we keep a stable version complete with all the right files on rachelfriends, so that we are not dependent on the ka-lite folks?

 

Bernd

 

.  <https://github.com/notifications/beacon/AMmq9DDJtDcaSzS5J4PlD65ysSLn0YUYks5ohDkagaJpZM4FOghH.gif>

**Comment by j-schwartz** [comment]
Hi Bernd,

I'd love to but it appears this new issue is related to taking the stable
exercises I have posted at rachelfriends and trying to use them in the new
0.14 kalite.

I think Ed has a workaround by adding 0.13 in an assessment items file, but
I need to check in the morning.

I'll spend Monday cleaning this up.

Thanks all

Jeremy

**Comment by bnordha** [comment]
Actually I pulled down the assessement.zip file ka-lite and had the same issue. That issue was also reported on ka-lite.

 

I posted the assessmentitems.json on our Github site and incorporated it into the install procedure.

 

The install process is working with the only exception that some video files are missing.

 

Bernd

**Comment by spatiald** [comment]
The install script downloads version 0.13.2 from the KA Lite GitHub.  That is the version I am running.

For reference, the config script:
  1. Installs KA Lite, if not installed; updated KA Lite, if it is installed
  2. Downloads/installs the assessmentitems.json that @bnordha uploaded to our GitHub
  3. Runs the setup_linux
  4. Configures the CONTENT_ROOT
  5. Downloads ka-lite_content.zip from rachelfriends.org and puts the content into the unzips /media/RACHEL/kacontent folder
  6. Starts KA Lite with the 20 sec pause at boot

@rachelproject/rachel I used the config script (based off of @bnordha original script) with success...once installed, I registered the device and then updated all the videos.  All the exercises I tested (included the ones @edresor mentioned above) work.  The videos work.  What exactly is not working for everyone?

Couple of other questions I am looking for answers to:

  1. @j-schwartz - Is there a index.php module built with the shell redirect to :8008 for KA Lite?
  2. Are we already sym linking the files and I don't know about it?
  3. Do we want to sym link them? (I think the answer is yes)

**Comment by j-schwartz** [comment]
I'm not sure what the remaining exercise issues are.  I had a good working
version two weeks ago, but I thought KA-lite updated to 0.14 and maybe that
was causing problems.  Strange that you're getting 0.13.2.  I'm wondering
if they reverted in someway.

You can grab the index.php with this: rsync -avz rsync://
dev.worldpossible.org/rachelmods/ka-lite ./

**Comment by hackersforcharity** [comment]
Thanks!  Only mod I didn't push yet to the config script is GCF.

On Jul 27, 2015, at 16:40, Jeremy Schwartz <notifications@github.com> wrote:

I'm not sure what the remaining exercise issues are. I had a good working
version two weeks ago, but I thought KA-lite updated to 0.14 and maybe that
was causing problems. Strange that you're getting 0.13.2. I'm wondering
if they reverted in someway.

You can grab the index.php with this: rsync -avz rsync://
dev.worldpossible.org/rachelmods/ka-lite ./

On Mon, Jul 27, 2015 at 1:34 PM Sam Kinch <notifications@github.com> wrote:

> The install script downloads version 0.13.2 from the KA Lite GitHub. That
> is the version I am running.
>
> For reference, the config script:
>
> 1. Installs KA Lite, if not installed; updated KA Lite, if it is
> installed
> 2. Downloads/installs the assessmentitems.json that @bnordha
> <https://github.com/bnordha> uploaded to our GitHub
> 3. Runs the setup_linux
> 4. Configures the CONTENT_ROOT
> 5. Downloads ka-lite_content.zip from rachelfriends.org and puts the
> content into the unzips /media/RACHEL/kacontent folder
> 6. Starts KA Lite with the 20 sec pause at boot
>
> @rachelproject/rachel <https://github.com/orgs/rachelproject/teams/rachel>
> I used the config script (based off of @bnordha
> <https://github.com/bnordha> original script) with success...once
> installed, I registered the device and then updated all the videos. All
the
> exercises I tested (included the ones @edresor
> <https://github.com/edresor> mentioned above) work. The videos work. What
> exactly is not working for everyone?
>
> Couple of other questions I am looking for answers to:
>
> 1. @j-schwartz <https://github.com/j-schwartz> - Is there a index.php
> module built with the shell redirect to :8008 for KA Lite?
>
>
> 1. Are we already sym linking the files and I don't know about it?
>
>
> 1. Do we want to sym link them? (I think the answer is yes)
>
>
>
> —
> Reply to this email directly or view it on GitHub
> <
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125334747>
> .
>

—
Reply to this email directly or view it on GitHub
<https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125336062>
.

**Comment by j-schwartz** [comment]
Great.  Somewhere along the way my content set of GCF was corrupted.  I'm
missing lots of content beyond the allsubjects.html page.   Do you have a
clean copy somewhere?

**Comment by bnordha** [comment]
The issue is with the missing assessmentitems.json file. It missing from both 0.13 and 0.14.

 

The file is supposed to be in ka-lite/data/khan but it is missing.

 

channel_data.json  exercises.json  topic_hierarchy.json

contents.json      images          topics.json

 

There are some entries on Github https://github.com/learningequality/ka-lite/issues/3898 and they closed it, but I tested it repeatedly and the issue still existis.

 

Bernd

**Comment by j-schwartz** [comment]
If you grab it from here and drop it in are you still having problems?
http://rachelfriends.org/z-holding/data-khan.zip

**Comment by spatiald** [comment]
The config script installs the assessmentitems.json file, wouldn't that
solve the issue?

On Jul 27, 2015, at 20:35, bnordha <notifications@github.com> wrote:

The issue is with the missing assessmentitems.json file. It missing from
both 0.13 and 0.14.



The file is supposed to be in ka-lite/data/khan but it is missing.



channel_data.json exercises.json topic_hierarchy.json

contents.json images topics.json



There are some entries on Github
https://github.com/learningequality/ka-lite/issues/3898 and they closed it,
but I tested it repeatedly and the issue still existis.



Bernd

**Comment by j-schwartz** [comment]
I think we're good. Talked to Bernd on Skype. Scripts are unreal great Sam.
Thanks.

**Comment by spatiald** [comment]
I uploaded the file GCFOffline.7z to dev.worldpossible.org

I just realized that 7za isn't on it to unzip.  I will need to re-upload or
have Jonathan install p7zip to uncompress.
sam

**Comment by needlestack** [comment]
Ok, installed p7zip on dev.worldpossible.org

Let me know what else you need!

**Comment by spatiald** [comment]
GCFOffline.7z can be downloaded from
dev.worldpossible.org/mods/GCFOffline.7z

The content is viewable at http://dev.worldpossible.org/mods/GCF/index.html

It is not rysnc-able yet...needs the htmlf filed created for it.

**Comment by j-schwartz** [comment]
I'll create today. I created a new kalite htmlf yesterday and I'm working
on ck12 today (updating all our textbooks with the update). I'll be putting
all the ck12 teachers editions in the Intel portal for teachers only
viewing.

Check out the latest kalite when you get a minute.

**Comment by needlestack** [comment]
Excellent, Sam - this is a very nice package.

I've added a basic index.htmlf earlier today. I just now saw your email
Jeremy - hope I caught you before you got to it! I also put up a .zip
version and added it all to the (increasingly unwieldy) modules page,
including the rsync command.

One thing to note about the GCF package is that it relies heavily on Flash.
It wouldn't be too much trouble to recompress the videos to .mp4, but it
also has a lot of interactive .swf files, some of which don't seem to
convert with Google's Swiffy Flash-to-HTML5 converter.

Cheers

**Comment by spatiald** [comment]
My understanding was that GCF was supposedly going to all HTML5.  I think they keep pushing their deadline but I will check in with them and find out for sure.

**Comment by spatiald** [comment]
I'll add it to the config script...yeah

On Aug 1, 2015, at 01:25, Jonathan Field <notifications@github.com> wrote:

Excellent, Sam - this is a very nice package.

I've added a basic index.htmlf earlier today. I just now saw your email
Jeremy - hope I caught you before you got to it! I also put up a .zip
version and added it all to the (increasingly unwieldy) modules page,
including the rsync command.

One thing to note about the GCF package is that it relies heavily on Flash.
It wouldn't be too much trouble to recompress the videos to .mp4, but it
also has a lot of interactive .swf files, some of which don't seem to
convert with Google's Swiffy Flash-to-HTML5 converter.

Cheers



On Fri, Jul 31, 2015 at 10:13 AM, Sam Kinch <notifications@github.com>
wrote:

> GCFOffline.7z can be downloaded from
> dev.worldpossible.org/mods/GCFOffline.7z
>
> The content is viewable at
> http://dev.worldpossible.org/mods/GCF/index.html
>
> It is not rysnc-able yet...needs the htmlf filed created for it.
>
> On Thu, Jul 30, 2015 at 2:48 AM, Jonathan Field <notifications@github.com>
> wrote:
>
>
> > Ok, installed p7zip on dev.worldpossible.org
> >
> > Let me know what else you need!
> >
> >
> > On Tue, Jul 28, 2015 at 4:34 PM, Sam Kinch <notifications@github.com>
> > wrote:
> >
> > > I uploaded the file GCFOffline.7z to dev.worldpossible.org
> > >
> > > I just realized that 7za isn't on it to unzip. I will need to
re-upload
> > or
> > > have Jonathan install p7zip to uncompress.
> > > sam
> > >
> > > On Mon, Jul 27, 2015 at 5:03 PM, Jeremy Schwartz <
> > notifications@github.com
> > > >
> > > wrote:
> > >
> > > > Great. Somewhere along the way my content set of GCF was corrupted.
> I'm
> > > > missing lots of content beyond the allsubjects.html page. Do you
> have a
> > > > clean copy somewhere?
> > > >
> > > >
> > > > On Mon, Jul 27, 2015 at 2:00 PM Hackers for Charity <
> > > > notifications@github.com> wrote:
> > > >
> > > > > Thanks! Only mod I didn't push yet to the config script is GCF.
> > > > >
> > > > > On Jul 27, 2015, at 16:40, Jeremy Schwartz <
> notifications@github.com
> > >
> > > > > wrote:
> > > > >
> > > > > I'm not sure what the remaining exercise issues are. I had a good
> > > working
> > > > > version two weeks ago, but I thought KA-lite updated to 0.14 and
> > maybe
> > > > that
> > > > > was causing problems. Strange that you're getting 0.13.2. I'm
> > wondering
> > > > > if they reverted in someway.
> > > > >
> > > > > You can grab the index.php with this: rsync -avz rsync://
> > > > > dev.worldpossible.org/rachelmods/ka-lite ./
> > > > >
> > > > > On Mon, Jul 27, 2015 at 1:34 PM Sam Kinch <
> notifications@github.com>
> > > > > wrote:
> > > > >
> > > > > > The install script downloads version 0.13.2 from the KA Lite
> > GitHub.
> > > > That
> > > > > > is the version I am running.
> > > > > >
> > > > > > For reference, the config script:
> > > > > >
> > > > > > 1. Installs KA Lite, if not installed; updated KA Lite, if it is
> > > > > > installed
> > > > > > 2. Downloads/installs the assessmentitems.json that @bnordha
> > > > > > <https://github.com/bnordha> uploaded to our GitHub
> > > > > > 3. Runs the setup_linux
> > > > > > 4. Configures the CONTENT_ROOT
> > > > > > 5. Downloads ka-lite_content.zip from rachelfriends.org and puts
> > the
> > > > > > content into the unzips /media/RACHEL/kacontent folder
> > > > > > 6. Starts KA Lite with the 20 sec pause at boot
> > > > > >
> > > > > > @rachelproject/rachel <
> > > > > https://github.com/orgs/rachelproject/teams/rachel>
> > > > > > I used the config script (based off of @bnordha
> > > > > > <https://github.com/bnordha> original script) with
> success...once
> > > > > > installed, I registered the device and then updated all the
> videos.
> > > All
> > > > > the
> > > > > > exercises I tested (included the ones @edresor
> > > > > > <https://github.com/edresor> mentioned above) work. The videos
> > work.
> > > > > What
> > > > > > exactly is not working for everyone?
> > > > > >
> > > > > > Couple of other questions I am looking for answers to:
> > > > > >
> > > > > > 1. @j-schwartz <https://github.com/j-schwartz> - Is there a
> > > index.php
> > > > > > module built with the shell redirect to :8008 for KA Lite?
> > > > > >
> > > > > >
> > > > > > 1. Are we already sym linking the files and I don't know about
> it?
> > > > > >
> > > > > >
> > > > > > 1. Do we want to sym link them? (I think the answer is yes)
> > > > > >
> > > > > >
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125334747
> > > > > >
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125336062
> > > > > >
> > > > > .
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125341276
> > > > >
> > > > > .
> > > > >
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125341936
> > > >
> > > > .
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125746236
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126203722
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126705530>
> .
>

—
Reply to this email directly or view it on GitHub
<https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126867102>
.

**Comment by hackersforcharity** [comment]
@anyone Dies anyone know the difference between the current kacontent and
those in Kaos?  There is about 23GB of duplicate content that I want to
clean up.
Sam


On Aug 2, 2015, at 08:01, Sam Kinch <notifications@github.com> wrote:

I'll add it to the config script...yeah

On Aug 1, 2015, at 01:25, Jonathan Field <notifications@github.com> wrote:

Excellent, Sam - this is a very nice package.

I've added a basic index.htmlf earlier today. I just now saw your email
Jeremy - hope I caught you before you got to it! I also put up a .zip
version and added it all to the (increasingly unwieldy) modules page,
including the rsync command.

One thing to note about the GCF package is that it relies heavily on Flash.
It wouldn't be too much trouble to recompress the videos to .mp4, but it
also has a lot of interactive .swf files, some of which don't seem to
convert with Google's Swiffy Flash-to-HTML5 converter.

Cheers



On Fri, Jul 31, 2015 at 10:13 AM, Sam Kinch <notifications@github.com>
wrote:

> GCFOffline.7z can be downloaded from
> dev.worldpossible.org/mods/GCFOffline.7z
>
> The content is viewable at
> http://dev.worldpossible.org/mods/GCF/index.html
>
> It is not rysnc-able yet...needs the htmlf filed created for it.
>
> On Thu, Jul 30, 2015 at 2:48 AM, Jonathan Field <notifications@github.com>
> wrote:
>
>
> > Ok, installed p7zip on dev.worldpossible.org
> >
> > Let me know what else you need!
> >
> >
> > On Tue, Jul 28, 2015 at 4:34 PM, Sam Kinch <notifications@github.com>
> > wrote:
> >
> > > I uploaded the file GCFOffline.7z to dev.worldpossible.org
> > >
> > > I just realized that 7za isn't on it to unzip. I will need to
re-upload
> > or
> > > have Jonathan install p7zip to uncompress.
> > > sam
> > >
> > > On Mon, Jul 27, 2015 at 5:03 PM, Jeremy Schwartz <
> > notifications@github.com
> > > >
> > > wrote:
> > >
> > > > Great. Somewhere along the way my content set of GCF was corrupted.
> I'm
> > > > missing lots of content beyond the allsubjects.html page. Do you
> have a
> > > > clean copy somewhere?
> > > >
> > > >
> > > > On Mon, Jul 27, 2015 at 2:00 PM Hackers for Charity <
> > > > notifications@github.com> wrote:
> > > >
> > > > > Thanks! Only mod I didn't push yet to the config script is GCF.
> > > > >
> > > > > On Jul 27, 2015, at 16:40, Jeremy Schwartz <
> notifications@github.com
> > >
> > > > > wrote:
> > > > >
> > > > > I'm not sure what the remaining exercise issues are. I had a good
> > > working
> > > > > version two weeks ago, but I thought KA-lite updated to 0.14 and
> > maybe
> > > > that
> > > > > was causing problems. Strange that you're getting 0.13.2. I'm
> > wondering
> > > > > if they reverted in someway.
> > > > >
> > > > > You can grab the index.php with this: rsync -avz rsync://
> > > > > dev.worldpossible.org/rachelmods/ka-lite ./
> > > > >
> > > > > On Mon, Jul 27, 2015 at 1:34 PM Sam Kinch <
> notifications@github.com>
> > > > > wrote:
> > > > >
> > > > > > The install script downloads version 0.13.2 from the KA Lite
> > GitHub.
> > > > That
> > > > > > is the version I am running.
> > > > > >
> > > > > > For reference, the config script:
> > > > > >
> > > > > > 1. Installs KA Lite, if not installed; updated KA Lite, if it is
> > > > > > installed
> > > > > > 2. Downloads/installs the assessmentitems.json that @bnordha
> > > > > > <https://github.com/bnordha> uploaded to our GitHub
> > > > > > 3. Runs the setup_linux
> > > > > > 4. Configures the CONTENT_ROOT
> > > > > > 5. Downloads ka-lite_content.zip from rachelfriends.org and puts
> > the
> > > > > > content into the unzips /media/RACHEL/kacontent folder
> > > > > > 6. Starts KA Lite with the 20 sec pause at boot
> > > > > >
> > > > > > @rachelproject/rachel <
> > > > > https://github.com/orgs/rachelproject/teams/rachel>
> > > > > > I used the config script (based off of @bnordha
> > > > > > <https://github.com/bnordha> original script) with
> success...once
> > > > > > installed, I registered the device and then updated all the
> videos.
> > > All
> > > > > the
> > > > > > exercises I tested (included the ones @edresor
> > > > > > <https://github.com/edresor> mentioned above) work. The videos
> > work.
> > > > > What
> > > > > > exactly is not working for everyone?
> > > > > >
> > > > > > Couple of other questions I am looking for answers to:
> > > > > >
> > > > > > 1. @j-schwartz <https://github.com/j-schwartz> - Is there a
> > > index.php
> > > > > > module built with the shell redirect to :8008 for KA Lite?
> > > > > >
> > > > > >
> > > > > > 1. Are we already sym linking the files and I don't know about
> it?
> > > > > >
> > > > > >
> > > > > > 1. Do we want to sym link them? (I think the answer is yes)
> > > > > >
> > > > > >
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125334747
> > > > > >
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125336062
> > > > > >
> > > > > .
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125341276
> > > > >
> > > > > .
> > > > >
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125341936
> > > >
> > > > .
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125746236
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126203722
> >
>
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126705530>
> .
>

—
Reply to this email directly or view it on GitHub
<https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126867102
>
.

—
Reply to this email directly or view it on GitHub
<https://github.com/rachelproject/rachelplus/issues/3#issuecomment-127019985>
.

**Comment by j-schwartz** [comment]
Ka content has more and a growing set of videos. I'm compressing more
videos now to add to ka content. I'd wait on messing around there until I
upload a new ka content. When uploaded, kacontent will have all videos
currently available for ka lite compressed.

**Comment by spatiald** [comment]
Do we all agree that /media/RACHEL/kacontent will be the only media
location for KA videos?  If so, we could copy all current videos there and
symlink all to cleanup. I believe an rsync with proper flags would only
update/newer videos but I should check on that...

On Aug 3, 2015, at 12:59, Jeremy Schwartz <notifications@github.com> wrote:

Ka content has more and a growing set of videos. I'm compressing more
videos now to add to ka content. I'd wait on messing around there until I
upload a new ka content. When uploaded, kacontent will have all videos
currently available for ka lite compressed.
On Mon, Aug 3, 2015 at 9:57 AM Hackers for Charity <notifications@github.com
>
wrote:

> @anyone Dies anyone know the difference between the current kacontent and
> those in Kaos? There is about 23GB of duplicate content that I want to
> clean up.
> Sam
>
>
> On Aug 2, 2015, at 08:01, Sam Kinch <notifications@github.com> wrote:
>
> I'll add it to the config script...yeah
>
> On Aug 1, 2015, at 01:25, Jonathan Field <notifications@github.com> wrote:
>
> Excellent, Sam - this is a very nice package.
>
> I've added a basic index.htmlf earlier today. I just now saw your email
> Jeremy - hope I caught you before you got to it! I also put up a .zip
> version and added it all to the (increasingly unwieldy) modules page,
> including the rsync command.
>
> One thing to note about the GCF package is that it relies heavily on
Flash.
> It wouldn't be too much trouble to recompress the videos to .mp4, but it
> also has a lot of interactive .swf files, some of which don't seem to
> convert with Google's Swiffy Flash-to-HTML5 converter.
>
> Cheers
>
>
>
> On Fri, Jul 31, 2015 at 10:13 AM, Sam Kinch <notifications@github.com>
> wrote:
>
> > GCFOffline.7z can be downloaded from
> > dev.worldpossible.org/mods/GCFOffline.7z
> >
> > The content is viewable at
> > http://dev.worldpossible.org/mods/GCF/index.html
> >
> > It is not rysnc-able yet...needs the htmlf filed created for it.
> >
> > On Thu, Jul 30, 2015 at 2:48 AM, Jonathan Field <
> notifications@github.com>
> > wrote:
> >
> >
> > > Ok, installed p7zip on dev.worldpossible.org
> > >
> > > Let me know what else you need!
> > >
> > >
> > > On Tue, Jul 28, 2015 at 4:34 PM, Sam Kinch <notifications@github.com>
> > > wrote:
> > >
> > > > I uploaded the file GCFOffline.7z to dev.worldpossible.org
> > > >
> > > > I just realized that 7za isn't on it to unzip. I will need to
> re-upload
> > > or
> > > > have Jonathan install p7zip to uncompress.
> > > > sam
> > > >
> > > > On Mon, Jul 27, 2015 at 5:03 PM, Jeremy Schwartz <
> > > notifications@github.com
> > > > >
> > > > wrote:
> > > >
> > > > > Great. Somewhere along the way my content set of GCF was
corrupted.
> > I'm
> > > > > missing lots of content beyond the allsubjects.html page. Do you
> > have a
> > > > > clean copy somewhere?
> > > > >
> > > > >
> > > > > On Mon, Jul 27, 2015 at 2:00 PM Hackers for Charity <
> > > > > notifications@github.com> wrote:
> > > > >
> > > > > > Thanks! Only mod I didn't push yet to the config script is GCF.
> > > > > >
> > > > > > On Jul 27, 2015, at 16:40, Jeremy Schwartz <
> > notifications@github.com
> > > >
> > > > > > wrote:
> > > > > >
> > > > > > I'm not sure what the remaining exercise issues are. I had a
good
> > > > working
> > > > > > version two weeks ago, but I thought KA-lite updated to 0.14 and
> > > maybe
> > > > > that
> > > > > > was causing problems. Strange that you're getting 0.13.2. I'm
> > > wondering
> > > > > > if they reverted in someway.
> > > > > >
> > > > > > You can grab the index.php with this: rsync -avz rsync://
> > > > > > dev.worldpossible.org/rachelmods/ka-lite ./
> > > > > >
> > > > > > On Mon, Jul 27, 2015 at 1:34 PM Sam Kinch <
> > notifications@github.com>
> > > > > > wrote:
> > > > > >
> > > > > > > The install script downloads version 0.13.2 from the KA Lite
> > > GitHub.
> > > > > That
> > > > > > > is the version I am running.
> > > > > > >
> > > > > > > For reference, the config script:
> > > > > > >
> > > > > > > 1. Installs KA Lite, if not installed; updated KA Lite, if it
> is
> > > > > > > installed
> > > > > > > 2. Downloads/installs the assessmentitems.json that @bnordha
> > > > > > > <https://github.com/bnordha> uploaded to our GitHub
> > > > > > > 3. Runs the setup_linux
> > > > > > > 4. Configures the CONTENT_ROOT
> > > > > > > 5. Downloads ka-lite_content.zip from rachelfriends.org and
> puts
> > > the
> > > > > > > content into the unzips /media/RACHEL/kacontent folder
> > > > > > > 6. Starts KA Lite with the 20 sec pause at boot
> > > > > > >
> > > > > > > @rachelproject/rachel <
> > > > > > https://github.com/orgs/rachelproject/teams/rachel>
> > > > > > > I used the config script (based off of @bnordha
> > > > > > > <https://github.com/bnordha> original script) with
> > success...once
> > > > > > > installed, I registered the device and then updated all the
> > videos.
> > > > All
> > > > > > the
> > > > > > > exercises I tested (included the ones @edresor
> > > > > > > <https://github.com/edresor> mentioned above) work. The videos
> > > work.
> > > > > > What
> > > > > > > exactly is not working for everyone?
> > > > > > >
> > > > > > > Couple of other questions I am looking for answers to:
> > > > > > >
> > > > > > > 1. @j-schwartz <https://github.com/j-schwartz> - Is there a
> > > > index.php
> > > > > > > module built with the shell redirect to :8008 for KA Lite?
> > > > > > >
> > > > > > >
> > > > > > > 1. Are we already sym linking the files and I don't know about
> > it?
> > > > > > >
> > > > > > >
> > > > > > > 1. Do we want to sym link them? (I think the answer is yes)
> > > > > > >
> > > > > > >
> > > > > > >
> > > > > > > —
> > > > > > > Reply to this email directly or view it on GitHub
> > > > > > > <
> > > > > >
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125334747
> > > > > > >
> > > > > > > .
> > > > > > >
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <
> > > > > >
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125336062
> > > > > > >
> > > > > > .
> > > > > >
> > > > > > —
> > > > > > Reply to this email directly or view it on GitHub
> > > > > > <
> > > > >
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125341276
> > > > > >
> > > > > > .
> > > > > >
> > > > >
> > > > > —
> > > > > Reply to this email directly or view it on GitHub
> > > > > <
> > > >
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125341936
> > > > >
> > > > > .
> > > > >
> > > >
> > > > —
> > > > Reply to this email directly or view it on GitHub
> > > > <
> > >
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-125746236
> > > >
> > >
> > > > .
> > > >
> > >
> > > —
> > > Reply to this email directly or view it on GitHub
> > > <
> >
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126203722
> > >
> >
> > > .
> > >
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126705530
> >
> > .
> >
>
> —
> Reply to this email directly or view it on GitHub
> <
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-126867102
> >
> .
>
> —
> Reply to this email directly or view it on GitHub
> <
>
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-127019985
> >
> .
>
> —
> Reply to this email directly or view it on GitHub
> <
https://github.com/rachelproject/rachelplus/issues/3#issuecomment-127330925>
> .
>

—
Reply to this email directly or view it on GitHub
<https://github.com/rachelproject/rachelplus/issues/3#issuecomment-127331282>
.

**Comment by j-schwartz** [comment]
Agreed.

**Comment by needlestack** [comment]
This sounds smart. Only thing is symlinks aren't very portable. It might be
wiser to make a second version of KAOS with corrected URLs? Let me know if
you want that and I'll throw it together.

I am curious about the future of KAOS - does it seem useful enough to
continue distribution (and even development)? I don't know how things pan
out vis-a-vis ka-lite...

**Comment by spatiald** [comment]
Could you set a variable in a config file that would point to the kacontent files location?  This way we can default to the USB setup for KAOS and then only need to change the config file for CAP, etc.

I do think KAOS is needed still.  KA Lite is definitely better than it used to be.  Could it replace KAOS at some point?  I guess…is the USB version of RACHEL running KA Lite?  I don’t think it is an option, is it?

sam

**Comment by spatiald** [comment]
Current USB Recovery has KA Lite installed with exercises but no video content.

The configure script will also install KA Lite with exercises plus allow the user to install the videos via two methods.  It will first attempt to download from rachel.golearn.us via rsync.  If that repo doesn't respond, it will try downloading the entire package from rachelfriends.org.  Both methods will continue syncing the videos/download where they left off if disconnected.  However, if you error out while pulling from the rachelfriends.org site, ensure you say 'Y' when it asks if there were errors so that your partial download doesn't get removed.  The script will then continue where it left off if re-run.  If we can get the video files copied up to rachelfriends.org, we may get faster sync times.  It would just be a matter of space on rachelfriends.org to store the videos. 

Thoughts?

**Comment by bnordha** [comment]
Sam

I actually just went through the process of downloading the videos. For me
the best is to rsync the videos outside of the installation process, and
then have the install do another rsync in case some videos were missed.

The reason that worked for me better was that downloading of the videos
takes hours, and because of that, chances that the rsync process gets
disrupted and I have to restart it are high.

It took me a while to figure out the correct rsync syntax (directories),
hence maybe you could  print out the rsync command, so users can copy and
paste it.

Does that make sense?

**Comment by spatiald** [comment]
The rsync command I use:

rsync -avhz --progress rsync://rachel.golearn.us/content/kacontent/ /media/RACHEL/kacontent/

It then fails over to download the entire video zip file.  Blah…

NEWLY ADDED, aka update your configure script - If the rsync process fails during install, you can run interactive mode, option 8 to retry.

sam

**Comment by needlestack** [comment]
2016.10.12 - buildRACHEL() process demonstrates how to get all kalite content in place - specifically using contentpacks, symlinking videos, and using a pre-built database that is now part of the kalite RACHEL modules.
